libdrgn/debug_info.c

```diff
@@ -2483,8 +2483,9 @@ parse_enumerator(Dwarf_Die *die, struct drgn_enum_type_builder *builder,
 					 "DW_TAG_enumerator is missing DW_AT_const_value");
 	}
 	struct drgn_error *err;
-	if (attr->form == DW_FORM_sdata ||
-	    attr->form == DW_FORM_implicit_const) {
+	// if (attr->form == DW_FORM_sdata ||
+	//     attr->form == DW_FORM_implicit_const) {
+	if (attr->form == DW_FORM_sdata) {
```

Comment:
This one I'm not as sure about. It feels wrong to ignore newer DWARF types even if the library doesn't support them. Maybe we should include copies of the latest `dwarf.h` and `known-dwarf.h` for drgn to use (since those just define constants) instead of the system ones?

Reply:
I agree that we should support newer DWARF types if we can.

As for the proposed solution, we don't have a lot of alternatives right? Maybe instead of a full copy of the header files we could only define that one constant for now? That said having the whole file may make more sense - we'd need to care less about whether new code may break old libelf users and pay closer attention to what constants drgn needs in general.

Reply:
Sure, for now you can just do

```c
#if !_ELFUTILS_PREREQ(0, 171)
#define DW_FORM_implicit_const 0x21
#endif
```

Probably put that next to the `dwelf_elf_begin()` stub. If we end up needing more of these, we can add the copy at that point.